includes/class-wc-gateway-stripe.php

```diff
@@ -841,6 +835,26 @@ public function process_payment( $order_id, $retry = true, $force_save_source =
 				$this->throw_localized_message( $response, $order );
 			}
 
+			if ( ! empty( $intent ) ) {
+				// Use the last charge within the intent to proceed.
+				$response = end( $intent->charges->data );
+
+				// If the intent requires a 3DS flow, redirect to it.
+				if ( 'requires_source_action' === $intent->status ) {
+					if ( is_wc_endpoint_url( 'order-pay' ) ) {
+						$redirect_url = add_query_arg( 'wc-stripe-confirmation', 1, $order->get_checkout_payment_url( false ) );
+					} else {
+						// $redirect_url = add_query_arg( 'wc-stripe-confirmation', 1, $order->get_checkout_payment_url( false ) );
```

Comment:
???

Reply:
This comment is a left-over, I performed multiple tests of the flow with Checkout + Pay for Order (instead of a modal shown based on hashes). Now it's removed.